swaggerPath.js

```diff
@@ -0,0 +1,251 @@
+ *                 startCity:
+ *                   type: string
+ *                 destination:
+ *                   type: string
+ *                 totalSeats:
+ *                   type: integer
+ *                 availableSeats:
+ *                   type: integer
+ *             example:
+ *               _id: "609170d64c9ac313185ddca4"
+ *               busNum: "B001"
+ *               busType: "Sleeper"
+ *               startCity: "Mumbai"
+ *               destination: "Pune"
+ *               totalSeats: 30
+ *               availableSeats: 30
+ *       '400':
+ *         description: Invalid request 
+ *       '401':
+ *         description: Unauthorized access
+ *       '500':
+ *         description: Internal server error
+ */
+
+
+/**
+ * @swagger
+ *
+ * /api/buses/{id}:
+ *   get:
+ *     summary: Get bus details by Id
+ *     tags: [Buses]
+ *     parameters:
+ *       - in: path
+ *         name: id
+ *         schema:
+ *           type: string
+ *         required: true
+ *         description: The Id of the bus to get details
+ *     responses:
+ *       200:
+ *         description: The bus details were successfully retrieved
+ *       404:
+ *         description: The bus with the specified ID was not found
+ */
+
+
+/**
+ * @swagger
+ * /api/buses/{id}:
+ *   put:
+ *     summary: Update bus details
+ *     description: Update the details of an existing bus
+ *     tags: [Buses] 
+ *     parameters:
+ *       - in: path
+ *         name: id
+ *         description: Id of the bus to update
+ *         required: true
+ *         schema:
+ *           type: string
+ *     requestBody:
+ *       description: New bus details to update
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               busNum:
+ *                 type: string
+ *               busType:
+ *                 type: string
+ *               startCity:
+ *                 type: string
+ *               destination:
+ *                 type: string
+ *               totalSeats:
+ *                 type: number
+ *               availableSeats:
+ *                 type: number
+ *     responses:
+ *       200:
+ *         description: Bus details updated successfully
+ *       404:
+ *         description: Bus not found
+ *       500:
+ *         description: Internal server error
+ */
+
+
+/**
+ * @swagger
+ *
+ * /api/buses/{id}:
+ *   delete:
+ *     summary: Delete bus details
+ *     description: Delete bus details based on ID
+ *     tags: [Buses] 
+ *     parameters:
+ *       - in: path
+ *         name: id
+ *         schema:
+ *           type: string
+ *         required: true
+ *         description: ID of the bus to delete
+ *     responses:
+ *       200:
+ *         description: The bus was successfully deleted
+ *       404:
+ *         description: The bus with the specified ID was not found
+ */
+
+
+
```

Comment:
Excellent documentation, testing, and coding formats,
Can restructure the file structure